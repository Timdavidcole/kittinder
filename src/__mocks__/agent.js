const Breeds = {
  all: jest.fn()
};

const Images = {
  get: jest.fn()
}

const Favourites = {
  post: jest.fn()
}

export default { Breeds, Images, Favourites };
