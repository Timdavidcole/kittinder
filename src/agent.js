import _superagent from "superagent";

const API_ROOT = "https://api.thecatapi.com/v1/";
const API_KEY = "a701ff7c-5713-41b4-b4e7-86671f2ce5a3";

const responseBody = res => res.body;

const requests = {
  get: (url, query) =>
    _superagent
      .get(`${API_ROOT}${url}`)
      .set("x-api-key", API_KEY)
      .query(query)
      .then(responseBody),
  post: (url, body) =>
    _superagent
      .post(`${API_ROOT}${url}`, body)
      .then(responseBody),
  put: (url, body) =>
    _superagent
      .put(`${API_ROOT}${url}`, body)
      .then(responseBody)
};

const Images = {
  all: (query) => requests.get(`images/search`, query),
  byAuthor: author =>
    requests.get(`/notices?author=${encodeURIComponent(author)}&limit=5`),
  favoritedBy: author =>
    requests.get(`/notices?favorited=${encodeURIComponent(author)}&limit=5`),
  get: slug => requests.get(`/notices/${slug}`),
  del: slug => requests.del(`/notices/${slug}`)
};

export default {Images};