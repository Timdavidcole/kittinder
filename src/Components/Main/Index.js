import React from "react";
import agent from "../../agent";
import CatCardContainer from "../CatCard/CatCardContainer";

class Index extends React.Component {
  constructor(props) {
    super(props);
    this.state = { error: true };
  }

  componentDidMount() {
    agent.Breeds.all()
      .then(data => {
        this.props.updateCats(data);
        this.setState({ error: false });
      })
      .catch(() =>
        this.setState({
          error: true
        })
      );
  }

  render() {
    if (this.state.error) {
      return (
        <div style={{ padding: "5px ", fontSize: "0.8rem" }}>
          The cats are a bit shy right now, please try again later.
        </div>
      );
    } else {
      return (
        <div>
          <CatCardContainer userId={this.props.userId} cats={this.props.cats} />
          ;
        </div>
      );
    }
  }
}

export default Index;
