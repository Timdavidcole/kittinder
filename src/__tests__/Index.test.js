import Index from "../Components/Main/Index";
import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

jest.mock("../agent");
import agent from "../agent";
jest.mock("../Components/CatCard/CatCardContainer");

beforeEach(() => {
  agent.Breeds.all.mockImplementation(() => {
    return Promise.resolve([
      { id: "test", name: "testcat", description: "Just a test cat" }
    ]);
  });
});

test("checks Index snapshot", async () => {
  const index = await renderer.create(<Index />);

  let tree = index.toJSON();
  expect(tree).toMatchSnapshot();
});

test("CatCard has correct breeds array from API call", async () => {
  const wrapper = await shallow(<Index />);
  expect(wrapper.state("cats")).toEqual([
    { id: "test", name: "testcat", description: "Just a test cat" }
  ]);
});

test("CatCard has correct breeds array from API call", async () => {
  agent.Breeds.all.mockImplementation(() => {
    return Promise.reject(new Error("fail"));
  });

  const wrapper = await shallow(<Index />);
  expect(wrapper.text()).toBe(
    "The cats are a bit shy right now, please try again later."
  );
});
