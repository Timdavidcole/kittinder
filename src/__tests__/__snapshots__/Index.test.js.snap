// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks Index snapshot 1`] = `
<div
  style={
    Object {
      "fontSize": "0.8rem",
      "padding": "5px ",
    }
  }
>
  The cats are a bit shy right now, please try again later.
</div>
`;
